kolibri/plugins/facility/assets/src/routes.js

```diff
@@ -24,21 +24,45 @@ import {
 } from './modules/classAssignMembers/handlers';
 import { PageNames } from './constants';
 
-function facilityParamRequiredGuard(toRoute, subtopicName) {
-  if (store.getters.userIsMultiFacilityAdmin && !toRoute.params.facility_id) {
-    router.replace({
-      name: 'ALL_FACILITIES_PAGE',
-      params: { subtopicName },
-    });
-    return true;
+function facilityParamRequiredGuard(toRoute, fromRoute, subtopicName) {
+  if (
+    store.getters.userIsMultiFacilityAdmin &&
+    !toRoute.params.facility_id &&
+    fromRoute.query.subtopicName !== subtopicName
+  ) {
+    const currentSubtopicName = toRoute.query.subtopicName;
+    if (currentSubtopicName !== subtopicName) {
+      router.replace({
+        name: PageNames.ALL_FACILITIES_PAGE,
+        query: { subtopicName },
+        params: { subtopicName },
+      });
+      return true;
+    }
+    return false;
+  } else {
+    if (!toRoute.params.facility_id) {
```

Comment:
This will catch the instance where the facility_id can indeed be optional - which is when the user is not a multi-facility admin, so this doesn't seem quite right. What behaviour is this meant to catch?